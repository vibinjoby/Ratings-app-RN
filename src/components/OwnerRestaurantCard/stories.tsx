import React from 'react'
import { storiesOf } from '@storybook/react-native'

import OwnerRestaurantCard from './component'

export const Story = () => (
  <OwnerRestaurantCard
    id={123}
    title={'Moxies Grill & Bar'}
    ratings={4.5}
    reviewsCount={4}
    onPress={() => console.log('pressed')}
  />
)

storiesOf('Components', module).add('OwnerRestaurantCard', () => <Story />)

export default {
  title: 'OwnerRestaurantCard',
}
