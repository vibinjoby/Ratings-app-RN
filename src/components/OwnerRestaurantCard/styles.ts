import { StyleSheet } from 'react-native'

import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'flex-start',
    marginRight: 10,
    backgroundColor: Colors.white,
    borderRadius: 8,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  img: {
    height: 165,
    borderTopEndRadius: 8,
    borderTopLeftRadius: 8,
  },
  title: {
    ...Typography.BoldFont,
    fontSize: 16,
    textAlign: 'center',
    marginVertical: 6,
  },
  footer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 15,
  },
  ratings: {
    fontSize: 13,
    color: Colors.lightText,
    marginHorizontal: 5,
  },
  reviews: {
    fontSize: 13,
    color: Colors.lightText,
    marginRight: 5,
  },
})
