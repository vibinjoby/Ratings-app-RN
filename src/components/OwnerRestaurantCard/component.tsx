import React from 'react'
import { Image, Text, TouchableOpacity, View } from 'react-native'

import styles from './styles'
import constants from '../../configs/commonConst'
import Stars from '../Stars'
import Colors from '../../utilities/colors'

export interface OwnerRestaurantCardProps {
  testID?: string
  id: number
  title: string
  ratings: number
  reviewsCount: number
  onPress: (id: number) => void
}

const OwnerRestaurantCard: React.FC<OwnerRestaurantCardProps> = ({
  id,
  testID,
  title,
  ratings,
  reviewsCount = 0,
  onPress,
}: OwnerRestaurantCardProps) => {
  return (
    <TouchableOpacity
      testID={testID}
      onPress={() => onPress(id)}
      style={styles.container}
      activeOpacity={0.7}
    >
      <Image style={styles.img} source={{ uri: constants.DUMMY_PIC }} />
      <Text testID="title" style={styles.title} numberOfLines={1}>
        {title}
      </Text>
      <View style={styles.footer}>
        <Text testID="ratings" style={styles.ratings}>
          {ratings}
        </Text>
        <Stars
          testID="stars"
          starHeight={10}
          selectedStars={ratings}
          selectedColor={Colors.appOrange}
        />
        <Text testID="reviewsCount" style={styles.reviews}>
          {reviewsCount} Reviews
        </Text>
      </View>
    </TouchableOpacity>
  )
}

export default OwnerRestaurantCard
