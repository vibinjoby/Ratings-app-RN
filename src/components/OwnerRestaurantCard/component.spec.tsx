import 'react-native'
import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import OwnerRestaurantCard from './component'

describe('OwnerRestaurantCard', () => {
  const MOCK_PRESS = jest.fn()
  const MOCK_TITLE = 'Moxies Grill & Bar'
  const MOCK_RATINGS = 4
  const MOCK_REVIEWS_COUNT = 4

  it('should render', () => {
    const { toJSON } = render(
      <OwnerRestaurantCard
        title={MOCK_TITLE}
        ratings={MOCK_REVIEWS_COUNT}
        reviewsCount={MOCK_RATINGS}
        onPress={MOCK_PRESS}
      />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render all values', () => {
    const { getByTestId } = render(
      <OwnerRestaurantCard
        testID="card"
        title={MOCK_TITLE}
        ratings={MOCK_REVIEWS_COUNT}
        reviewsCount={MOCK_RATINGS}
        onPress={MOCK_PRESS}
      />,
    )
    expect(getByTestId('title').props.children).toBe(MOCK_TITLE)
    expect(getByTestId('ratings').props.children).toBe(MOCK_RATINGS)
  })

  it('should be clickable', () => {
    const { getByTestId } = render(
      <OwnerRestaurantCard
        testID="card"
        title={MOCK_TITLE}
        ratings={MOCK_REVIEWS_COUNT}
        reviewsCount={MOCK_RATINGS}
        onPress={MOCK_PRESS}
      />,
    )
    fireEvent.press(getByTestId('card'))
    expect(MOCK_PRESS).toBeCalled()
  })
})
