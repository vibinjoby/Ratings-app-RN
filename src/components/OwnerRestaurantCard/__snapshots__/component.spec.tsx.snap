// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnerRestaurantCard should render 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "elevation": 5,
      "flex": 1,
      "marginRight": 10,
      "opacity": 1,
      "shadowColor": "#181725",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://source.unsplash.com/user/picoftasty",
      }
    }
    style={
      Object {
        "borderTopEndRadius": 8,
        "borderTopLeftRadius": 8,
        "height": 165,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "fontFamily": "Poppins-Bold",
        "fontSize": 16,
        "marginVertical": 6,
        "textAlign": "center",
      }
    }
    testID="title"
  >
    Moxies Grill & Bar
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(60, 60, 67, 0.6)",
          "fontSize": 13,
          "marginHorizontal": 5,
        }
      }
      testID="ratings"
    >
      4
    </Text>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          1,
          2,
          3,
          4,
          5,
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="stars0"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FF5823",
                    "fontSize": 10,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="font-icon 0"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginRight": 5,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="stars1"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FF5823",
                    "fontSize": 10,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="font-icon 1"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginRight": 5,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="stars2"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FF5823",
                    "fontSize": 10,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="font-icon 2"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginRight": 5,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="stars3"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FF5823",
                    "fontSize": 10,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="font-icon 3"
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginRight": 5,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="stars4"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#A5A6B7",
                    "fontSize": 10,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="font-icon 4"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <Text
      style={
        Object {
          "color": "rgba(60, 60, 67, 0.6)",
          "fontSize": 13,
          "marginRight": 5,
        }
      }
      testID="reviewsCount"
    >
      4
       Reviews
    </Text>
  </View>
</View>
`;
